avro-fastserde/src/main/java/com/linkedin/avro/fastserde/primitive/PrimitiveArrayList.java

```diff
@@ -0,0 +1,137 @@
+package com.linkedin.avro.fastserde.primitive;
+
+import java.util.AbstractList;
+import java.util.Iterator;
+import org.apache.avro.generic.GenericArray;
+import org.apache.avro.generic.GenericContainer;
+import org.apache.avro.generic.GenericData;
+
+
+public abstract class PrimitiveArrayList<T, L> extends AbstractList<T>
+    implements GenericContainer, Comparable<GenericArray<T>> {
+  private final Class<L> type;
+  protected int size = 0;
+
+  public PrimitiveArrayList(Class<L> type) {
+    this.type = type;
+  }
+
+  @Override public int size() {
+    return size;
+  }
+
+  @Override public void clear() {
+    size = 0;
+  }
+
+  @Override public Iterator<T> iterator() {
+    return new Iterator<T>() {
+      private int position = 0;
+
+      @Override
+      public boolean hasNext() {
+        return position < size;
+      }
+
+      @Override
+      public T next() {
+        return get(position++);
+      }
+
+      @Override
+      public void remove() {
+        throw new UnsupportedOperationException();
+      }
+    };
+  }
+
+  protected abstract int capacity();
+
+  protected void checkIfLargerThanSize(int i) {
+    if (i >= size)
```

Comment:
please always use {}

Reply:
Will do. That's my preference as well, actually... this code is mostly copied from `GenericData.Array`, which has this unsavory coding style xD ... I'll fix it.